system/middleware.go

```diff
@@ -41,6 +42,45 @@ func (application *Application) ApplyDbMap(c *web.C, h http.Handler) http.Handle
 	return http.HandlerFunc(fn)
 }
 
+func (application *Application) ApplyAPIAuth(c *web.C, h http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		if c.Env["IsAPI"].(bool) {
+			var apitoken string
+
+			authHeader := r.Header.Get("Authorization")
+			if strings.HasPrefix(authHeader, "Bearer ") {
+				apitoken = strings.TrimPrefix(authHeader, "Bearer ")
+
+				JWTtoken, err := jwt.Parse(apitoken, func(token *jwt.Token) (interface{}, error) {
+					// validate signing algorithm
+					if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+						return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+					}
+					return []byte(application.APISecret), nil
+				})
+
+				if err != nil {
+					log.Warnf("invalid token %v: %v", apitoken, err)
+				} else {
+					if claims, ok := JWTtoken.Claims.(jwt.MapClaims); ok && JWTtoken.Valid {
+						dbMap := c.Env["DbMap"].(*gorp.DbMap)
```

Comment:
Yeah, I don't know why it didn't bother me before.  Maybe because there are more middleware with a required order.  Whatever the reason I think there should be a note about the required order somewhere